ql/src/java/org/apache/hadoop/hive/ql/DriverContext.java

```diff
@@ -125,6 +130,12 @@ public QueryPlan getPlan() {
 
   public void setPlan(QueryPlan plan) {
     this.plan = plan;
+    TezTask task = Utilities.getFirstTezTask(plan.getRootTasks()).orElse(null);
```

Comment:
@abstractdog  After query run completes, when resources are being released by the driver, plan is being set to null and it is causing NullPointerException here, which is present as part of DEBUG logs.

Reply:
I see, can you point me to the logs that show this?

Reply:
`0: jdbc:hive2://localhost:10000> show tables;
INFO : Compiling command(queryId=tanishqchugh_20241113082511_604beac2-0686-4de3-93e2-edb71629232c): show tables
INFO : Semantic Analysis Completed (retrial = false)
INFO : Created Hive schema: Schema(fieldSchemas:[FieldSchema(name:tab_name, type:string, comment:from deserializer)], properties:null)
INFO : Completed compiling command(queryId=tanishqchugh_20241113082511_604beac2-0686-4de3-93e2-edb71629232c); Time taken: 0.071 seconds
INFO : Executing command(queryId=tanishqchugh_20241113082511_604beac2-0686-4de3-93e2-edb71629232c): show tables
INFO : PREHOOK: query: show tables
INFO : PREHOOK: type: SHOWTABLES
INFO : PREHOOK: Input: database:default
INFO : Starting task [Stage-0:DDL] in serial mode
DEBUG : Task getting executed using mapred tag : tanishqchugh_20241113082511_604beac2-0686-4de3-93e2-edb71629232c,userid=tanishqchugh
INFO : POSTHOOK: query: show tables
INFO : POSTHOOK: type: SHOWTABLES
INFO : POSTHOOK: Input: database:default
INFO : Completed executing command(queryId=tanishqchugh_20241113082511_604beac2-0686-4de3-93e2-edb71629232c); Time taken: 0.105 seconds
DEBUG : Exception while clearing the Fetch task
java.lang.NullPointerException: null
	at org.apache.hadoop.hive.ql.DriverContext.setPlan(DriverContext.java:133) ~[hive-exec-4.1.0-SNAPSHOT.jar:4.1.0-SNAPSHOT]
	at org.apache.hadoop.hive.ql.Driver.releasePlan(Driver.java:794) ~[hive-exec-4.1.0-SNAPSHOT.jar:4.1.0-SNAPSHOT]
	at org.apache.hadoop.hive.ql.Driver.releaseResources(Driver.java:611) ~[hive-exec-4.1.0-SNAPSHOT.jar:4.1.0-SNAPSHOT]
	at org.apache.hadoop.hive.ql.Driver.runInternal(Driver.java:214) ~[hive-exec-4.1.0-SNAPSHOT.jar:4.1.0-SNAPSHOT]
	at org.apache.hadoop.hive.ql.Driver.run(Driver.java:143) ~[hive-exec-4.1.0-SNAPSHOT.jar:4.1.0-SNAPSHOT]
	at org.apache.hadoop.hive.ql.Driver.run(Driver.java:138) ~[hive-exec-4.1.0-SNAPSHOT.jar:4.1.0-SNAPSHOT]
	at org.apache.hadoop.hive.ql.reexec.ReExecDriver.run(ReExecDriver.java:190) ~[hive-exec-4.1.0-SNAPSHOT.jar:4.1.0-SNAPSHOT]
	at org.apache.hive.service.cli.operation.SQLOperation.runQuery(SQLOperation.java:234) ~[hive-service-4.1.0-SNAPSHOT.jar:4.1.0-SNAPSHOT]
	at org.apache.hive.service.cli.operation.SQLOperation.access$500(SQLOperation.java:88) ~[hive-service-4.1.0-SNAPSHOT.jar:4.1.0-SNAPSHOT]
	at org.apache.hive.service.cli.operation.SQLOperation$BackgroundWork$1.run(SQLOperation.java:334) ~[hive-service-4.1.0-SNAPSHOT.jar:4.1.0-SNAPSHOT]
	at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_422]
	at javax.security.auth.Subject.doAs(Subject.java:422) ~[?:1.8.0_422]
	at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1899) ~[hadoop-common-3.3.6.jar:?]
	at org.apache.hive.service.cli.operation.SQLOperation$BackgroundWork.run(SQLOperation.java:354) ~[hive-service-4.1.0-SNAPSHOT.jar:4.1.0-SNAPSHOT]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_422]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_422]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[?:1.8.0_422]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[?:1.8.0_422]
	at java.lang.Thread.run(Thread.java:750) ~[?:1.8.0_422]
DEBUG : Shutting down query show tables
+------------------+
|   tab_name   |
+------------------+
| customer     |
| employees    |
| final_employees |
| lineitem     |
| nation      |
| orders      |
| part       |
| partsupp     |
| region      |
| supplier     |
| temp       |
+------------------+
11 rows selected (0.258 seconds)
0: jdbc:hive2://localhost:10000>`

Reply:
Sure
[temp.txt](https://github.com/user-attachments/files/17745294/temp.txt)

Reply:
this shouldn't cause any issue wrt query execution but just some noise in logs. a null check wrt plan should just be fine.. i guess the one test failure in precommit is as well linked to this ;)